backend/services/music_gen_service/music_tools.py

```
@@ -0,0 +1,1872 @@
+            "tempo": tempo,
+            "key": key,
+            "time_signature": self.default_time_signature,
+            "instruments": instruments
+        }
+    
+    async def _generate_midi(self, music_description: Dict[str, Any]) -> Dict[str, Any]:
+        """Generate MIDI files from the music description and return a detailed JSON."""
+        logger.debug(f"Generating MIDI for: {music_description.get('title', 'Untitled')}")
+        
+        # Generate separate MIDI files
+        results = await midi_generator.generate_midi_separate(music_description)
```

Ruff (0.8.2)

Undefined name `midi_generator`

(F821)